vertx-web/src/test/java/io/vertx/ext/web/handler/sse/SSEBaseTest.java

```diff
@@ -0,0 +1,204 @@
+  protected final String EB_ADDRESS = "eb-forward-sse";
+  protected String multilineMsg = "Some message split across multiple lines";
+  protected final static String SSE_NO_CONTENT_ENDPOINT = "/sse-no-content";
+  protected final static String SSE_REJECT_ODDS = "/sse-reject-odds";
+  protected final static String SSE_RESET_CONTENT_ENDPOINT = "/sse-reset-content";
+  protected final static String SSE_RECONNECT_ENDPOINT = "/sse-reconnect";
+  protected final static String SSE_REDIRECT_ENDPOINT = "/sse-redirect";
+  protected final static String SSE_EVENTBUS_ENDPOINT = "/sse-eventbus";
+  protected final static String SSE_MULTIPLE_MESSAGES_ENDPOINT = "/sse-multiple-messages";
+  protected final static String SSE_ID_TEST_ENDPOINT = "/sse-id-tests";
+  protected final static String SSE_MULTILINE_ENDPOINT = "/sse-multiline";
+  protected final static String SSE_ENDPOINT = "/sse";
+
+  private final static Integer PORT = 9009;
+
+  protected SSEConnection connection;
+
+  private HttpServer server;
+  private HttpClientOptions options;
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    CountDownLatch latch = new CountDownLatch(1);
+    HttpServerOptions options = new HttpServerOptions();
+    options.setPort(PORT);
+    server = vertx.createHttpServer(options);
+    Router router = Router.router(vertx);
+    // Non-SSE handlers
+    router.get(SSE_NO_CONTENT_ENDPOINT).handler(rc -> rc.response().setStatusCode(204).end());
+    router.get(SSE_RESET_CONTENT_ENDPOINT).handler(rc -> rc.response().setStatusCode(205).end());
+    router.get(SSE_REDIRECT_ENDPOINT).handler(rc -> {
+      // Client-side redirect
+      rc.response().putHeader(HttpHeaders.LOCATION.toString(), SSE_ENDPOINT + "?token=" + TOKEN);
+      rc.response().setStatusCode(302).end();
+    });
+    // Proper SSE handlers
+    attachSSEHandlers(router);
+    server.requestHandler(router);
+    server.listen(ar -> {
+      if (ar.failed()) {
+        fail(ar.cause());
+      }
+      latch.countDown();
+    });
+    awaitLatch(latch);
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    super.tearDown();
+    connection = null;
+  }
+
+  EventSource eventSource(long retryPeriod) {
+    return EventSource.create(vertx, clientOptions().setRetryPeriod(retryPeriod));
+  }
+
+  EventSource eventSource() {
+    return EventSource.create(vertx, clientOptions());
+  }
+
+  HttpClient client() {
+    return vertx.createHttpClient(clientOptions());
+  }
+
+  private EventSourceOptions clientOptions() {
+    if (options == null) {
+      options = new HttpClientOptions();
+      options.setDefaultPort(PORT);
+    }
+    return new EventSourceOptions(options);
+  }
+
+  private void attachSSEHandlers(Router router) {
+    // A simple SSE handler that doesn't do anything specific apart accepting the connection
+    SSEHandler sseHandler = SSEHandler.create();
+    sseHandler.connectHandler(connection -> {
+      this.connection = connection; // accept
```

Comment:
Why do we need this (from an API perspective)? Access Control should be done at a previous handler IMHO (unless this isn't for access control)